### ISSUE TITLE
Dictionaries allow string keys with number as first character, but produce error when accessed

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot#43607]
**Godot version:**
3.2.3 stable

**OS/device including version:**
Windows 10

**Issue description:**
(Actually, I'm not sure if this is a bug or intended behavior, considering variable names can't begin with a number)

Godot will let you create a dictionary with string keys beginning with numbers without complaint.
These keys will even show up as autocomplete results, but trying to access the associated value will give a confusing error.
(In this case, it says the line needs to end with a colon, even though it already does.)

**Steps to reproduce:**
Create a `String` key with a number as the first character.
![DictKeyBeginningWithStringNumber](https://user-images.githubusercontent.com/60025044/99354220-aaf92b00-286b-11eb-8be7-76195e016589.png)

**The error-producing code:**
```gdscript
const Directions := {
	"2D": [Vector2.LEFT, Vector2.RIGHT, Vector2.UP, Vector2.DOWN],
	"3D": [Vector3.LEFT, Vector3.RIGHT, Vector3.UP, Vector3.DOWN],
}

func _ready():
	for __2D_direction in Directions.2D:
		print(__2D_direction)
```

**Minimal reproduction project:**
[DictKeyBeginningWithStringNumber.zip](https://github.com/godotengine/godot/files/5551648/DictKeyBeginningWithStringNumber.zip)

**Comment by hilfazer** [comment]
String starting with digits are one of those Strings where you can't skip quotes. Same thing in Lua-style dictionaries:

`var lua = { 2s = 9 } # error`

or with $ syntax sugar:

`$2bar # error`

Use Dictionary's syntax ( `dict["string"]` ).

**Comment by Error7Studios** [comment]
Yes, that makes the code work, but the error message still doesn't make any sense.
I'm not sure if it's possible to do, but it would be much more helpful if the error message was more like this one.

![DictKeyBeginningWithStringNumber2](https://user-images.githubusercontent.com/60025044/99357855-42fa1300-2872-11eb-8d8e-813581273125.png)

Something like:
`Syntax error: (Directions.2D) Did you mean Directions["2D"]?`
